src/renderer/api/kube-object.ts

```diff
@@ -184,6 +223,12 @@ export class KubeObject implements ItemObject {
 
   // use unified resource-applier api for updating all k8s objects
   async update<T extends KubeObject>(data: Partial<T>) {
+    for (const precondition of KubeObject.editPreconditions) {
```

Comment:
What if preconditions for some objects would be different (extended)? 
How we would overcome this in `update()`?

Reply:
We would override this method in the child classes, doing those pre-condition checks and then calling `super.update(data)`;

I could see an argument for using a registry instead (if for some reason some of these base checks are wrong for some object types). But I think doing that now is overkill.

Finally, it should be noted that `kubectl` does neither of those and merely checks all kube objects (though it doesn't block updating the `resourceVersion`, I blocked that because it fixes a bug where if you do that only the edited item is left in the store visually).